core/src/main/kotlin/com/samples/verifier/internal/SamplesVerifierInstance.kt

```diff
@@ -95,26 +117,29 @@ internal class SamplesVerifierInstance(compilerUrl: String, kotlinEnv: KotlinEnv
     val fileTree = directory.walkTopDown()
       .onEnter {
         (configuration.ignoreDirectory == null ||
-          configuration.ignoreDirectory?.matches(it.relativeTo(directory).toString()) != true)
+          configuration.ignoreDirectory?.matches(it.toRelativeString(directory)) != true)
       }
     for (file in fileTree) {
       val dir = file.relativeTo(directory).toString()
       if (fileRegex?.matches(dir) == false) continue
-      snippets.addAll(processFile(file, type))
+      snippets.addAll(processFile(directory, file, type))
     }
     return snippets
   }
 
-  private fun processFiles(directory: File, filenames: List<String>, type: FileType): List<CodeSnippet> {
+  private fun filterFiles(filenames: List<String>): List<String> {
     val fileRegex = configuration.parseDirectory?.separatePattern()
     val ignoreRegex = configuration.ignoreDirectory?.separatePattern()
     return filenames
-      .filter { fileRegex?.matches(it) != false && ignoreRegex?.matches(it) != true }
-      .map { File(it) }
-      .flatMap { processFile(directory.resolve(it), type) }
+      .filter { fileRegex?.matches(it) ?: true && !(ignoreRegex?.matches(it) ?: false) }
+  }
+
+  private fun processFiles(directory: File, filenames: List<String>, type: FileType): List<CodeSnippet> {
+    return filterFiles(filenames).map { File(it) }
+      .flatMap { processFile(directory, directory.resolve(it), type) }
   }
 
-  private fun processFile(file: File, type: FileType): List<CodeSnippet> {
+  private fun processFile(baseDir: File, file: File, type: FileType): List<CodeSnippet> {
     return when (type) {
```

Comment:
I guess you can reuse processFile(filename: String, content: String, type: FileType) here

Reply:
No, ```processFile(file: File, type: FileType)``` reads from the file implicitly , but it needs to check file extension before. A call ```processFile(file.toRelativeString(baseDir), file.readText(), type)``` is senselessly.